UrlShortener/src/main/java/matthewgroves/urlShortener/resources/UrlShortenerResource.java

```diff
@@ -81,20 +112,24 @@ public Response expandUrl(@PathParam("id") String id) {
 	@Timed
 	@Consumes(MediaType.TEXT_PLAIN)
 	@Produces(MediaType.APPLICATION_JSON)
-	// TODO - This should return a POJO, not a Response
-	public Response addShortenedUrl(String fullUrl) {
-		try {
-			if (fullUrl != null && fullUrl.length() > 0 && fullUrl.length() < 2048) {
-				long id = dao.insertUrl(fullUrl);
-				
-				ShortenedUrl sUrl = new ShortenedUrl(id, fullUrl, "http://" + config.getHostname() + ":"
-						+ config.getRuntimePort() + "/shrtn/" + Long.toString(id, 36));
-				return Response.ok().entity(sUrl).build();
-			}
-		} catch (Exception e) {
-			return Response.serverError().entity(e).build();
-		}
+	public ShortenedUrl addShortenedUrl(String fullUrl) {
+		if (fullUrl == null || fullUrl.length() == 0 || fullUrl.length() > 2048)
```

Comment:
FYI: Dropwizard has a validation layer too http://www.dropwizard.io/0.9.0/docs/manual/validation.html#validations.

So you could write:
`@NotEmpty @Length(max=2048) String fullUrl`
but then you wouldn't be able to test the class standalone.